packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -94,92 +89,80 @@ export default class RequestClient {
-        }
-        this.preflightDone = true
-        return this.allowedHeaders.slice()
-      })
-      .catch((err) => {
-        this.uppy.log(`[CompanionClient] unable to make preflight request ${err}`, 'warning')
-        this.preflightDone = true
-        return this.allowedHeaders.slice()
-      })
+    const allowedHeadersNew = await promise
+    const fallbackAllowedHeaders = ['accept', 'content-type', 'uppy-auth-token']
+    return allowedHeadersNew != null ? allowedHeadersNew : fallbackAllowedHeaders
   }
 
-  preflightAndHeaders (path) {
-    return Promise.all([this.preflight(path), this.headers()])
-      .then(([allowedHeaders, headers]) => {
-        // filter to keep only allowed Headers
-        Object.keys(headers).forEach((header) => {
-          if (!allowedHeaders.includes(header.toLowerCase())) {
-            this.uppy.log(`[CompanionClient] excluding disallowed header ${header}`)
-            delete headers[header] // eslint-disable-line no-param-reassign
-          }
-        })
-
-        return headers
-      })
-  }
+  async preflightAndHeaders (path) {
+    const [allowedHeaders, headers] = await Promise.all([this.preflight(path), this.headers()])
+    // filter to keep only allowed Headers
+    Object.fromEntries(Object.entries(headers).filter(([header]) => {
+      if (!allowedHeaders.includes(header.toLowerCase())) {
+        this.uppy.log(`[CompanionClient] excluding disallowed header ${header}`)
+        return false
+      }
+      return true
+    }))
 
-  get (path, skipPostResponse) {
-    const method = 'get'
-    return this.preflightAndHeaders(path)
-      .then((headers) => fetchWithNetworkError(this.#getUrl(path), {
-        method,
-        headers,
-        credentials: this.opts.companionCookiesRule || 'same-origin',
-      }))
-      .then(this.#getPostResponseFunc(skipPostResponse))
-      .then(handleJSONResponse)
-      .catch(this.#errorHandler(method, path))
+    return headers
   }
 
-  post (path, data, skipPostResponse) {
-    const method = 'post'
-    return this.preflightAndHeaders(path)
-      .then((headers) => fetchWithNetworkError(this.#getUrl(path), {
-        method,
-        headers,
-        credentials: this.opts.companionCookiesRule || 'same-origin',
-        body: JSON.stringify(data),
-      }))
-      .then(this.#getPostResponseFunc(skipPostResponse))
-      .then(handleJSONResponse)
-      .catch(this.#errorHandler(method, path))
-  }
+  get = async (path, skipPostResponse) => this.#request({ path, skipPostResponse })
+
+  post = async (path, data, skipPostResponse) => this.#request({ path, method: 'POST', data, skipPostResponse })
 
-  delete (path, data, skipPostResponse) {
-    const method = 'delete'
-    return this.preflightAndHeaders(path)
-      .then((headers) => fetchWithNetworkError(`${this.hostname}/${path}`, {
+  delete = async (path, data, skipPostResponse) => this.#request({ path, method: 'DELETE', data, skipPostResponse })
```

Comment:
```suggestion
  async delete (path, data, skipPostResponse) { return this.#request({ path, method: 'DELETE', data, skipPostResponse }) }
```

Reply:
what's the purpose of this? seems more verbose and less readable to me

Reply:
```js
class MyClass {
  property = () => {}
  method() {}
}
console.log(new MyClass().method === new MyClass().method);
console.log(new MyClass().property !== new MyClass().property);
```

Using a class field will redefine and assign the function to each instance when calling `new MyClass`, while using a method only define the function once, and it's shared between all instances.